hnn_core/optimization.py

```diff
@@ -351,24 +439,29 @@ def optimize_evoked(params, target_dpl, initial_dpl, maxiter=50,
     smooth_window_len : int
         The length of the hamming window (in samples) to smooth the
         simulated dipole waveform in each optimization step.
+    dt : float
+        The integration time step (ms) of h.CVode during simulation.
 
     Returns
     -------
-    params : dict
-        The optimized params dictionary.
+    net : Instance of Network object
+        Network instance with the optimized configuration of attached drives.
 
     Notes
     -----
     This optimization protocol utilizes the Constrained Optimization
     By Linear Approximation (COBYLA) method:
     https://docs.scipy.org/doc/scipy/reference/generated/scipy.optimize.fmin_cobyla.html  # noqa
     """
-    from .parallel_backends import _BACKEND, JoblibBackend
 
-    if _BACKEND is None:
-        _BACKEND = JoblibBackend(n_jobs=1)
+    drive_names = [key for key in net.external_drives.keys()
+                   if net.external_drives[key]['type'] == 'evoked']
+    if len(drive_names) == 0:
+        raise ValueError(f'The current Network instance lacks any evoked '
+                         f'drives. Got {net.external_drives}')
```

Comment:
```suggestion
                         f'drives.')
```

The "got" part is unhelpful here because the message will always read:

> The current Network instance lacks any evoked drives. Got []

Instead maybe you can say:

> The current Network instance lacks any evoked drives. Consider adding drives using net.add_evoked_drive

Reply:
Oh boy - not my brightest moment. Good catch!